src/agent.js

```diff
@@ -20,10 +20,23 @@ const onAuthorized = ( { socket, events } ) => ( agent ) => {
 		debug( 'received message', message )
 		events.emit( 'message', message )
 	} )
+
+	socket.on( 'system.info', done => {
+		events.emit( 'system.info', data => done( data ) )
+	} )
+
 	socket.on( 'role.add', ( role, done ) => {
 		debug( 'agent joining role', role )
+		events.emit( 'role.add', role )
 		socket.join( role, done )
 	} )
+
+	socket.on( 'disconnect', () => {
+		for ( const room in socket.rooms ) {
+			events.emit( 'role.remove', room )
```

Comment:
Added these events so `happychat` can listen for leave and remove the interval when the room is empty.